app/src/main/java/com/example/weather/ui/main/MainActivity.java

```diff
@@ -11,23 +12,22 @@
 import androidx.appcompat.app.AppCompatActivity;
 
 import com.example.weather.R;
+import com.example.weather.data.RepositoryImpl;
 import com.example.weather.databinding.ActivityMainBinding;
+import com.example.weather.utils.Constant;
 import com.example.weather.utils.SettingManager;
 import com.google.android.material.dialog.MaterialAlertDialogBuilder;
 
 public class MainActivity extends AppCompatActivity {
 
     ActivityMainBinding binding;
-    SettingManager settingManager;
+    SettingManager settingManager= RepositoryImpl.getInstance().getSettingsMenager();
```

Comment:
No, you didn't catch. It should be only like a method inside `ViewModel`. So, SettingManager's instance inside repository and repository use methods (`setLocale(...)`), `ViewModel` has instance of the `Repository` and invokes `setLocale(...)` and then you can invoke this method via `ViewModel`. This logic is similar with usage of network, db and other services. We shouldn't depend on realisation.